integration-libs/punchout/root/config/punchout-guard-config.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Config } from '@spartacus/core';
+import { LaunchRoute } from '@spartacus/storefront';
+import { PunchOutOperation } from '../model';
+
+@Injectable({
+  providedIn: 'root',
+  useExisting: Config,
+})
+export abstract class PunchoutNavigationGuardConfig {
+  punchoutNavigation?: {
```

Comment:
In my understanding, punchoutNavigation property is needed as we are using dynamic keys (eg:[PunchOutOperation.EDIT]) and those cannot be at root level.